cascade-prelude/src/Cascade/Data/Text/NonEmpty.hs

```diff
@@ -32,3 +38,17 @@ pattern NonEmpty a <- Mk a
 
 mk :: Text -> Maybe NonEmpty
 mk input = if Text.null input then Nothing else Just $ Mk input
+
+data ValidationError = IsEmpty
+  deriving stock (Generic, Data, Show)
+  deriving ToJSON via (ApiErrorFormat ValidationError)
```

Comment:
This data structure should not be aware about API. Actually this type is an internal type which should not be able to reflect on contracts. Let's defer instance derivation to the `cascade-api` package in a newtype wrapping this one.